.github/workflows/sonatype_scan.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. 
+# SPDX-License-Identifier: Apache-2.0
+# Licensed under the Apache License, Version 2.0 https://aws.amazon.com/apache-2-0/
+
+name: Sonatype SCA Scanning
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  SonatypeUrl: "https://finos.sonatype.app/platform/"
+  SonatypeAppId: "htc-grid"
+  SonatypeScanTarget: "."
+  ExcludeDirectory: "**/docs/ **/examples/ **/.*'"
+
+jobs:
+  build:
+    if: github.repository_owner == 'finos'
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip' # caching pip dependencies
+      
+      - name: Download Python Libraries
+        run : |
+          projRoot=`pwd`
+          list=$(find "." -name 'requirements.txt')
+          
+          for i in $list; do
+              echo ""
+              echo "Foud requirements.txt at: $i"
+              j="${i/'requirements.txt'/}" # Remove 'requirements.txt' from path
+              cd $j
+
+              # Remove local dependencies
+              sed -i '/dist/d' requirements.txt 
+              sed -i '/backports.zoneinfo/d' requirements.txt
+              
+              cat requirements.txt
+              echo ""
+              mkdir pip-modules
+              pip install -r requirements.txt
```

Comment:
Why are we installing them if we are downloading them? Looks like this should have commented out as well, together with freeze? And, if we are going to install and freeze them, then we should probably do this inside a `virtualenv` rather than directly on host as there may then be potential conflicts between the different versions of a package needed by different components.

Reply:
@fgogolli the install of the dependencies is to resolve blank versions or version ranges. The dependencies should be scoped to that directory level of the project and we are just using it to scan the dependencies.